src/runtime/contrib/tensorrt/tensorrt_builder.cc

```diff
@@ -208,8 +208,35 @@ runtime::TrtEngineAndContext TensorRTBuilder::BuildEngine(
   CHECK_EQ(engine->getNbBindings(),
            num_input_bindings + network_output_names_.size());
   nvinfer1::IExecutionContext* context = engine->createExecutionContext();
-  return {engine, context, network_input_names_, network_input_is_baked_,
-          network_output_names_};
+  std::vector<runtime::NDArray> device_buffers;
+  for (int i = 0; i < network_input_names_.size(); i++) {
+    if (network_input_is_baked_[i]) continue;
+    int binding_index = engine->getBindingIndex(network_input_names_[i].c_str());
+    if (execution_args_[binding_index]->ctx.device_type == kDLGPU) {
```

Comment:
Use `execution_args_[i]`.
`execution_args_` has a different ordering than bindings/buffers, so binding_index is not correct. Can you add a second test case with one input which is a param, so that we can test that the indexing is correct.